### ISSUE TITLE
Auto-renewable subscription doesn't renew on Sandbox

### ISSUE THREAD

**Issue by enrickdl** [iaphub/react-native-iaphub#55]
Hey guys, how's everything?

I'm setting up IAPHUB on my React Native App and using Webhooks on a node backend. Yesterday, while testing, I noticed that all my purchases weren't being renewed, even though all my subscriptions are set as Auto-renewable subscription. When I click on the webhook to see more information, I've noticed this field on the request body:

`"subscriptionCancelReason":"customer_cancelled"`

The funny thing is that I haven't canceled the subscription at any anytime and throughout the last week, I wasn't getting this kind of behavior, all my subscriptions were auto-renewing. Is this normal when I'm using the Sandbox environment? Or it can be something on my app or our backend? Again, this just started happening yesterday.

Thanks in advance!

**Comment by iaphub** [comment]
Hi @enrickdl,

It is normal, itunes renew a subscription in sandbox up to 5 times and cancels the subscription randomly for testing purposes.
If you try buying a subscription later again you'll probably have renewals.

**Comment by enrickdl** [comment]
Hi @iaphub, that wasn't the case. The problem was: I'm buying an auto-renewable subscription for the first time and it expires with 5 minutes, without even trying to renew as it happens with sandbox.

However, I find out that if I'm not logged in to App Store, all my purchases will be expired, so I believe that this was the problem, since I had to switch Apple Id accounts and the App Store doesn't automatically log in with the new account.

**Comment by enrickdl** [comment]
Hey @iaphub, Unfortunately, I was wrong and the problem still persists. I've tried with all my accounts logged (Apple Id and App Store) and the same error happening. 

The first try that I had today worked (My comment 3 hours ago), but all the other ones didn't. I've tried with different uses and the same problem persists.

I'll attach a photo of all of my requests today:

![Screen Shot 2021-01-08 at 11 32 57](https://user-images.githubusercontent.com/65981081/104026845-4a200c00-51a5-11eb-8309-945cb48e7457.png)


---------------------


Another problem is that I've got 4 different products (2 with a free trial and 2 without), and when I create a listing, the `subscriptionPeriodType` shows accordingly. But after the first requests, all of the products show `subscriptionPeriodType: normal` even though 2 of them should show `trial`. Should I open a new issue for this?
